### PR TITLE
Feature/microvg platformdevguide

### PR DESCRIPTION
**Pull request by gbalan-microej**
* see result here: https://docs.microej.com/en/feature-microvg_platformdevguide/PlatformDeveloperGuide/vg.html
* missing a chapter on the VG images / image generator (optional)
* the images are wrong (I prefer to wait for your feedback first)

### CODE REVIEW

File: PlatformDeveloperGuide/vgReleaseNotes.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+* ``LLVG_PATH_PAINTER_impl.h`` (see Path module's :ref:`section_vg_path_llapi`) and ``microvg_path.h`` (see :ref:`section_vg_c_module_microvg`): the painter implementation for NXP i.MX RT500 (that provides Vivante VGLITE GPU),
+* ``LLVG_FONT_PAINTER_impl.h`` (see Font module's :ref:`section_vg_font_llapi`): the font painter implementation for NXP i.MX RT500 (that provides Vivante VGLITE GPU).
+
+This module requires the C module `MicroUI over VGLite`: `<zzz_devrepourl_zzz/com/microej/clibrary/llimpl/microui-vglite>`_
+
+Compatibility
+-------------
+
+The C module `MicroVG-VGLite` fetches automatically by transitivity the other C modules.
+No need to fetch explicitly the other modules.
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/vgReleaseNotes.rst:82]: harfbuzz module has to be fetch as it is a optional module.

File: PlatformDeveloperGuide/vgReleaseNotes.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+Several C modules are available on `MicroEJ Central Repository <zzz_centralrepourl_zzz/modules>`_  and `Developer Repository <zzz_devrepourl_zzz/microej-developer-repository-release>`_
+
+.. _section_vg_c_module_microvg:
+
+MicroVG
+-------
+
+The C module `<zzz_centralrepourl_zzz/modules/com/microej/clibrary/llimpl/microvg>`_ provides default implementations for:
+
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/vgReleaseNotes.rst:52]: Should we talk about the microvg_configuration.h file here?

File: PlatformDeveloperGuide/index.rst [hunk 3]
```diff
@@ -20,6 +20,7 @@ Platform Developer Guide
    externalResourceLoader
    serialCommunications
    ui
+   vg
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/index.rst:23]: I don't see any reference to the microvg.properties file.

File: PlatformDeveloperGuide/vgMatrix.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* matrix_memory[6] = matrix[2][0];
+* matrix_memory[7] = matrix[2][1];
+* matrix_memory[8] = matrix[2][2];
+
+.. _section_vg_matrix_llapi:
+
+Low Level API
+=============
+
+The Low Level APIs to implement are listed in the header file ``LLVG_MATRIX_impl.h`` (see :ref:`LLVG-MATRIX-API-SECTION`):
```
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/vgMatrix.rst:38]: no. I list the API in the appendices

File: PlatformDeveloperGuide/appendix/llapi.rst [hunk 5]
```diff
@@ -429,6 +429,188 @@ This UI extension provides support to efficiently implement the set of methods t
...
+In case of the buffer is too small to add the new command, the implementation has to return a value that indicates the number of bytes the array must be enlarged. 
+
+List of commands:
+
+* ``LLVG_PATH_CMD_CLOSE``: MicroVG "CLOSE" command.
+* ``LLVG_PATH_CMD_MOVE``: MicroVG "MOVE ABS" command.
+* ``LLVG_PATH_CMD_MOVE_REL``: MicroVG "MOVE REL" command.
+* ``LLVG_PATH_CMD_LINE``: MicroVG "LINE ABS" command.
+* ``LLVG_PATH_CMD_LINE_REL``: MicroVG "LINE REL" command.
+* ``LLVG_PATH_CMD_QUAD``: MicroVG "QUAD EBS" command.
```
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/appendix/llapi.rst:513]: arg copy/paste from C doc! I have to fix the C doc...

File: PlatformDeveloperGuide/vgGradient.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+   Gradient Low Level API
+
+Use
+===
+
+The MicroVG Gradient APIs are available in the class `ej.microvg.LinearGradient <zzz_javadocurl_zzz/ej/microvg/LinearGradient.html>`_.
+
+..
+   | Copyright 2008-2022, MicroEJ Corp. Content in this space is free
```
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/vgGradient.rst:48]: i don't know -> asked to #_doc

File: PlatformDeveloperGuide/vgReleaseNotes.rst (hunk 2 above)
Comment (gbalan-microej) [reply PlatformDeveloperGuide/vgReleaseNotes.rst:52]: you're right: see last commit

File: PlatformDeveloperGuide/index.rst (hunk 3 above)
Comment (gbalan-microej) [reply PlatformDeveloperGuide/index.rst:23]: You're right. This file is mandatory (I have tested without it: it does not work). It should be available in configuration/microvg. However this considers the vg-pack' scripts are executed during platform build (there is a script that copy user's file in platform folder during the platform build). This feature is not currently supported in PCA (branch not merged). 

I have two solutions;
- ask to put this file in configuration/dropins/microvg
- change the script to use "vglite" by default (the other implementations are not fully done yet)

File: PlatformDeveloperGuide/vgGradient.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/vgGradient.rst:49]: spec: use 2008-2022 everywhere

File: PlatformDeveloperGuide/vgReleaseNotes.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The C module `MicroVG-VGLite` fetches automatically by transitivity the other C modules.
+No need to fetch explicitly the other modules.
+An update of this C module updates (if necessary) the other C modules.
+The following table describes the compatibility versions between this C module and the MicroEJ VG Packs:
+
++-----------------------------+---------------+
+| MicroVG-VGLite Module Range | VG Pack Range |
++=============================+===============+
+| 3.0.0                       | 1.1.0         |
```
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/vgReleaseNotes.rst]: I have delivered a new chapter "C Modules" and updated the release notes

File: PlatformDeveloperGuide/vgCco.rst [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+Principle
+=========
+
+Several C modules implement the VG Pack's Low-Level APIs.
+Some are generic, and some are platform-specific dependent (more precisely: GPU dependent).
+The generic modules provide header files to be extended by the specific modules. 
+The generic C modules are available on `MicroEJ Central Repository <zzz_centralrepourl_zzz/modules>`_  and the specific C modules on the `Developer Repository <zzz_devrepourl_zzz/microej-developer-repository-release>`_.
+
+The following picture illustrates the available C modules and their relations.
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/vgCco.rst:15]: Explain that this picture is for an implementation over vglite :
"The following picture illustrates the available C modules and their relations for an implementation that uses:
- freetype library for the font renderer and the font layouter in simple layout mode.
- harfbuzz library for the font layouter in complex layout mode.
- Vivante VGLite library for the drawing of vector paths

File: PlatformDeveloperGuide/vgCco.rst [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+It implements a set of drawings over the official Vivante VGLite library (that targets some GPU with vector graphics acceleration): ``drawing_vglite.c``.
+
+This C module also provides a set of header files (and their implementations) to manipulate some MicroUI concepts over the VGLite library: image management, path format, etc.: ``display_vglite.h`` and ``vglite_path.h``.
+
+This C module is available on the Developer Repository: XXXX_zzzz_link.
+
+Library: Vivante VGLite
+=======================
+
+This library symbolizes the official Vivante VGLite library.
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/vgCco.rst:58]: I don't really like the term "symbolizes". Maybe just "This library is the official Vivante VGLite library" would be ok.

File: PlatformDeveloperGuide/vgCco.rst [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+The C modules use its header files to target the GPU.
+
+.. note:: The library must be patched to be compatible with the C module "MicroUI over VGLite". Consult the C module's ReadMe file to have more information.
```
Comment (arouaux-mej) [inline PlatformDeveloperGuide/vgCco.rst:61]: to have -> for

File: PlatformDeveloperGuide/index.rst (hunk 3 above)
Comment (arouaux-mej) [inline PlatformDeveloperGuide/index.rst:23]: Images review:
- some arrow are over boxes on some images.
- I have see the term "CCO" in an image. Do we want to keep it? Or do we want to call this C Module?
- Do we need to specify the version on MICROVG (1.1) in all drawings?

Comment (gbalan-microej) [reply PlatformDeveloperGuide/index.rst:23]: * arrows: this is the expected format: can require a full block or just a component and always implement a component...
* cco: you're right: replaces by "c module"
* version of MicroVG: as MicroUI actually but I agree: I remove it

**Pull request merged**